build-gcc-ibex.sh

```diff
@@ -18,6 +18,7 @@ TOOLCHAIN_NAME=lowrisc-toolchain-gcc-rv32imc-$TAG_NAME
 # config.
 #export CT_PREFIX=/tools/riscv
 
+echo CT_LOCAL_PATCH_DIR=\"$PWD/patches\" | tee -a lowrisc-toolchain-gcc-rv32imc.config
```

Comment:
You do that for an absolute path in the config? You can avoid that by using `CT_LOCAL_PATCH_DIR=${CT_TOP_DIR}/patches` directly in the config, and store the the patches in a folder `patches` relative to where you call ct-ng.

Reply:
Please don't. It's easier to keep the two locations in sync if it's done this way.

Reply:
Why is it harder to keep the two in sync then?

Reply:
That's what I was doing originally, but @lenary suggested the current approach to avoid copying the patches to `${CT_TOP_DIR}/patches` and to ensure the path wouldn't change. I suppose a middle-ground would be to use something like `CT_LOCAL_PATCH_DIR=${CT_TOP_DIR}/../../patches`. I think @lenary's point is that it wouldn't be obvious you have to change that setting if you changed the paths elsewhere. I don't have strong opinions about this.

Reply:
I was never a fan of people modifying the ct-ng config with text editing tools (you see a lot of sed and friends out there), as it makes it harder to use these configs without the magic machinery around it, and it also keeps the config now effectively in two places.

Since we already copy the config around into another directory, I don't see a problem with symlinking/copying the patches directory as well.

Either way, not blocking on that.

Reply:
I feel there is a fundamental difference between tee-ing a few variables onto the end of a config file, and using something like sed, especially if the original config file documents which variables will be tee'd onto the end.

Reply:
It leads roughly to the same outcome: a config which doesn't work without being modified.

Reply:
I don't feel there's a difference between "read the config comments and add this line" vs "read the config and know where to move your patches dir". Importantly in the first one, the location of the patches dir is originally specified in only one place, rather than having to update two places when the repo is inevitable reorganised.

Reply:
From my point of view, I don't have strong feelings about generating configuration files. But I do believe that it's wrong for a script to edit a file that's checked in! If I understand correctly, running `build-gcc-ibex.sh` will modify `lowrisc-toolchain-gcc-rv32imc.config`. Yuck. Surely, it's better to do something like `echo whatever | cat lowrisc-toolchain-gcc-rv32imc.config - >build/gcc/.config`, no?

Reply:
Yeah, @rswarbrick that would be better. Thanks!